sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py

```diff
@@ -228,12 +454,85 @@ def from_connection_string(cls, conn_str, **kwargs):
                 :dedent: 4
                 :caption: Create a new instance of the EventHubProducerClient from connection string.
         """
-        constructor_args = cls._from_connection_string(conn_str, **kwargs)
+        constructor_args = cls._from_connection_string(
+            conn_str,
+            eventhub_name=eventhub_name,
+            buffered_mode=buffered_mode,
+            on_success=on_success,
+            on_error=on_error,
+            max_buffer_length=max_buffer_length,
+            max_wait_time=max_wait_time,
+            **kwargs
+        )
         return cls(**constructor_args)
 
+    def send_event(self, event_data, **kwargs):
+        # type: (Union[EventData, AmqpAnnotatedMessage], Any) -> None
+        """
+        Sends an event data.
+        By default, the method will block until acknowledgement is received or operation times out.
+        If the `EventHubProducerClient` is configured to run in buffered mode, the method will enqueue the event
+        into local buffer and return. The producer will do automatic batching and sending in the background.
+
+        If `on_error` is passed while `buffered_mode` is False when instantiating the producer client,
+        instead of error being raised from the send methods,
+        the `on_error` callback will be called with the error information related to sending.
+
+        :param event_data: The `EventData` object to be sent.
+        :type event_data: Union[~azure.eventhub.EventData, ~azure.eventhub.amqp.AmqpAnnotatedMessage]
+        :keyword float timeout: The maximum wait time to send the event data.
+         If not specified, the default wait time specified when the producer was created will be used.
+        :keyword str partition_id: The specific partition ID to send to. Default is None, in which case the service
+         will assign to all partitions using round-robin.
+         A `TypeError` will be raised if partition_id is specified and event_data_batch is an `EventDataBatch` because
+         `EventDataBatch` itself has partition_id.
+        :keyword str partition_key: With the given partition_key, event data will be sent to
+         a particular partition of the Event Hub decided by the service.
+         A `TypeError` will be raised if partition_key is specified and event_data_batch is an `EventDataBatch` because
+         `EventDataBatch` itself has partition_key.
+         If both partition_id and partition_key are provided, the partition_id will take precedence.
+         **WARNING: Setting partition_key of non-string value on the events to be sent is discouraged
+         as the partition_key will be ignored by the Event Hub service and events will be assigned
+         to all partitions using round-robin. Furthermore, there are SDKs for consuming events which expect
+         partition_key to only be string type, they might fail to parse the non-string value.**
```

Comment:
We should add an exception tag:
`:raises OperationTimeoutError: If the value specified by the timeout parameter elapses before the event can be sent and not running in buffered mode.`
Otherwise we can add this to the description for the timeout parameter itself.

Reply:
Same comment applies to `send_batch`

Reply:
updated!